xarray/core/variable.py

```diff
@@ -1212,34 +1212,79 @@ def __setitem__(self, key, value):
         raise TypeError('%s values cannot be modified' % type(self).__name__)
 
     @classmethod
+    def _concat_numpy(cls, variables, positions=None):
+        """
+        Concatenates variables. Works for variables whose dtype is
+        different from numpy.object. If variables' dtype is numpy.object
+        it throws TypeError and "concat" function will use
+        concat_pandas function
+        :param variables: list of variables to concatenate
+        :return: Concatenated variables
+        """
+
+        indexes = [v._data for v in variables]
+
+        if not indexes:
```

Comment:
We don't need this special branch for numpy. Literally `return np.concatenate([v.data for v in variables])` would do for this function.